addons/mail/controllers/thread.py

```diff
@@ -163,3 +164,35 @@ def mail_message_update_content(self, message_id, body, attachment_ids, attachme
 
     def _is_message_editable(self, message, **kwargs):
         return message.sudo().is_current_user_or_guest_author or request.env.user._is_admin()
+
+    @http.route("/mail/message/get_followers", methods=["POST"], type="json", auth="public")
+    def message_get_followers(self, thread_id, thread_model, limit=20, offset=0, filter_recipients=False):
+        domain = [
+            ("res_id", "=", thread_id),
+            ("res_model", "=", thread_model),
+            ("partner_id", "!=", request.env.user.partner_id.id),
+        ]
+        if filter_recipients:
+            subtype_id = request.env["ir.model.data"]._xmlid_to_res_id("mail.mt_comment")
+            subtype_domain = [
+                ("subtype_ids", "=", subtype_id),
+                ("partner_id.active", "=", True),
+            ]
+            domain = expression.AND([domain, subtype_domain])
+        followersCount = request.env["mail.followers"].search_count(
+                    [("res_id", "=", thread_id), ("res_model", "=", thread_model)]
+                )
+        self_follower = request.env["mail.followers"].search(
+            [
+                ("res_id", "=", thread_id),
+                ("res_model", "=", thread_model),
+                ["partner_id", "=", request.env.user.partner_id.id],
+            ]
+        )
+        res = request.env["mail.followers"].search(domain, offset=offset, limit=limit, order="name ASC")
+        return {
+            "data": Store(res).get_result(),
+            "followers": Store.many_ids(res),
+            "selfFollower": Store(self_follower).get_result(),
+            "followersCount": followersCount
+            }
```

Comment:
Pass only a single `.get_result()`: this contains all the data to feed JS with `this.store.insert()`.

`get_result()` solves a problem to normalize and minimize data being passed, so any redudancies are avoided. By passing 2 `get_result()`, you pass redundant data thus leading to more data to process which not ok.

Prefer passing only id data in `followers` and `selfFollower`, so that you do `this.store.insert(data)`, but then rely on data in `this.store` to retrieve the records of `followers` and `selfFollower` thanks to the id data they contain.